packages/discovery-provider/src/queries/get_track_signature.py

```diff
@@ -164,13 +172,13 @@ def get_track_download_signature(args: GetTrackDownloadSignature):
     user_signature = args.get("user_signature")
     nft_access_signature = args.get("nft_access_signature")
     cid = track.get("orig_file_cid") if is_original else track.get("track_cid")
-    if not cid:
+    if not cid or not track.get("is_downloadable", False):
```

Comment:
Added this to the downloads checking. Verified on client we don't show the download item unless it's downloadable

Reply:
nice